### PR TITLE
several smaller improvements, plus idea for an alternative approach

### PR DESCRIPTION
**Pull request by drandreaskrueger**
Thank you so much, I was very happy to find your code. Well done. Fixed some minor issues. Might not be final yet, as I am trying out IPTV for the first time, and each new playlist ... comes up with new ways to break Kodi, lol. Best regards, Andreas.

### CODE REVIEW

**Comment by FutureSharks** [comment]
Hi @drandreaskrueger, thanks for such a great contribution!

I'm happy to merge it, just let me know when this PR is ready?

BTW, I haven't used this in over 2 years now. I tried it for a while but the streams were never reliable enough IMO, it was always a bit of a pain. But perhaps I'll give it another try 🙂

**Comment by drandreaskrueger** [comment]
It is ~ ready. It successfully reduces 765 entries to 499 which then seem all work. But I have not had the time to flick through really every resulting channel yet, so there might be some minor glitches left. But then I can simply do another PR. So, let's call this ready for now.

Before you merge, please go through my commits once. Because I might not have fully understood your program logic?  
For easier "quality control" I have kept every fix or feature in a separate commit - that should help you to quickly asses it.

Yes, do give it another try. For better IPTV user experience it makes all the difference that all(most all) channels are working.  
Thanks a lot again, great script.

File: TODO.md
Comment (FutureSharks) [inline TODO.md:36]: Good idea. Would be even better if it was possible to do it in python without using `subprocess` etc.

**Review by FutureSharks — APPROVED** [review]

**Comment by FutureSharks** [comment]
> Before you merge, please go through my commits once. Because I might not have fully understood your program logic?

Done, all good.

Thanks @drandreaskrueger, feel free to make further PRs 😃

**Pull request merged**
